sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -220,6 +222,404 @@ case class EndsWith(left: Expression, right: Expression)
+        data(i + j) = padTxt(j)
+        j += 1
+      }
+      i += pad.length()
+    }
+
+    // Copy the text
+    i = 0
+    while (pos + i < len && i < str.length()) {
+      data(pos + i) = txt(i)
+      i += 1
+    }
+
+    UTF8String.fromBytes(data)
+  }
+
+  override def toString: String = s"LPAD($str, $len, $pad)"
+}
+
+/**
+ * Returns str, right-padded with pad to a length of len.
+ */
+case class StringRPad(str: Expression, len: Expression, pad: Expression)
+  extends Expression with ExpectsInputTypes {
+
+  override def children: Seq[Expression] = str :: len :: pad :: Nil
+  override def foldable: Boolean = children.forall(_.foldable)
+  override def nullable: Boolean = children.exists(_.nullable)
+  override def dataType: DataType = StringType
+  override def expectedChildTypes: Seq[DataType] = Seq(StringType, IntegerType, StringType)
+
+  override def eval(input: InternalRow): Any = {
+    val s = str.eval(input)
+    if (s == null) {
+      null
+    } else {
+      val l = len.eval(input)
+      if (l == null) {
+        null
+      } else {
+        val p = pad.eval(input)
+        if (p == null) {
+          null
+        } else {
+          val len = l.asInstanceOf[Int]
+          val str = s.asInstanceOf[UTF8String]
+          val pad = p.asInstanceOf[UTF8String]
+          val bytes = new Array[Byte](len)
+
+          performOp(bytes, str.getBytes, pad.getBytes, len, str, pad)
+        }
+      }
+    }
+  }
+
+  // Copied from org.apache.hadoop.hive.ql.udf.generic.GenericUDFRpad
+  // ('hi', 5, '??') => 'hi???'
+  // ('hi', 1, '??') => 'h'
+  protected def performOp(
+    data: Array[Byte],
+    txt: Array[Byte],
+    padTxt: Array[Byte],
+    len: Int,
+    str: UTF8String,
+    pad: UTF8String): UTF8String = {
+
+    // Copy the text
+    var pos = 0
+    while (pos < str.length() && pos < len) {
+      data(pos) = txt(pos)
+      pos += 1
+    }
+
+    // Copy the padding
+    while (pos < len) {
+      var i = 0
+      while (i < pad.length() && i < len - pos) {
+        data(pos + i) = padTxt(i)
+        i += 1
+      }
+
+      pos += pad.length()
+    }
+
+    UTF8String.fromBytes(data)
+  }
+
+  override def toString: String = s"RPAD($str, $len, $pad)"
+}
+
+/**
+ * Returns the input formatted according do printf-style format strings
+ */
+case class StringFormat(children: Expression*) extends Expression {
+
+  require(children.length >=1, "printf() should take at least 1 argument")
+
+  override def foldable: Boolean = children.forall(_.foldable)
+  override def nullable: Boolean = children(0).nullable
+  override def dataType: DataType = StringType
+  private def format: Expression = children(0)
+  private def args: Seq[Expression] = children.tail
+
+  override def eval(input: InternalRow): Any = {
+    val pattern = format.eval(input)
+    if (pattern == null) {
+      null
+    } else {
+      val sb = new StringBuffer()
+      val formatter = new java.util.Formatter(sb, Locale.US)
```

Comment:
Actually I copied the idea from https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFPrintf.java#L100.
It sounds reasonable to me if we make it configurable, but I'd like to leave it for the next PR, as it probably break the unit test if people come from the non the US region.